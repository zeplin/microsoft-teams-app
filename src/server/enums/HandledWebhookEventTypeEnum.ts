export enum HandledWebhookEventTypeEnum {
    PING = "ping",
    PROJECT_COLOR = "project.color",
    PROJECT_NOTE = "project.note",
    PROJECT_NOTE_COMMENT = "project.note.comment",
    PROJECT_SPACING_TOKEN = "project.spacing_token",
    PROJECT_TEXT_STYLE = "project.text_style",
    PROJECT_MEMBER = "project.member",
    PROJECT_COMPONENT = "project.component",
    PROJECT_SCREEN = "project.screen",
    PROJECT_SCREEN_VERSION = "project.screen.version",
    STYLEGUIDE_COLOR = "styleguide.color",
    STYLEGUIDE_TEXT_STYLE = "styleguide.text_style",
    STYLEGUIDE_SPACING_TOKEN = "styleguide.spacing_token",
    STYLEGUIDE_MEMBER = "styleguide.member",
    STYLEGUIDE_COMPONENT = "styleguide.component",
}

export function isHandledWebhookEventTypeEnum(value: string): value is HandledWebhookEventTypeEnum {
    return Object.values<string>(HandledWebhookEventTypeEnum).includes(value);
}
