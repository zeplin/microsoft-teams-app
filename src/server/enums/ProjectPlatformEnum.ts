export enum ProjectPlatformEnum {
    ANDROID = "android",
    IOS = "ios",
    WEB = "web",
    MAC_OS = "macos",
}
