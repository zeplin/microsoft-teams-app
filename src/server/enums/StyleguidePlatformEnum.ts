export enum StyleguidePlatformEnum {
    BASE = "base",
    ANDROID = "android",
    IOS = "ios",
    WEB = "web",
    MAC_OS = "macos",
}
