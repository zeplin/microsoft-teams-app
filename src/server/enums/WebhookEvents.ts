export enum ProjectWebhookEvent {
    PROJECT_ALL = "project.*",
    PROJECT_COLOR = "project.color",
    PROJECT_COMPONENT = "project.component",
    PROJECT_MEMBER = "project.member",
    PROJECT_NOTE = "project.note",
    PROJECT_NOTE_COMMENT = "project.note.comment",
    PROJECT_SCREEN = "project.screen",
    PROJECT_SCREEN_VERSION = "project.screen.version",
    PROJECT_SPACING_TOKEN = "project.spacing_token",
    PROJECT_TEXT_STYLE = "project.text_style",
}
