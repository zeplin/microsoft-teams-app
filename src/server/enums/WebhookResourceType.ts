export enum WebhookResourceType {
    PROJECT = "Project",
    STYLEGUIDE = "Styleguide"
}
