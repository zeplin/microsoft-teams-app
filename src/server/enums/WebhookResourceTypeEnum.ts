export enum WebhookResourceTypeEnum {
    PROJECT = "Project",
    STYLEGUIDE = "Styleguide"
}
