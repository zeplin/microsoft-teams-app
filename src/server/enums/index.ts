export * from "./WebhookResourceTypeEnum";
export * from "./HandledWebhookEventTypeEnum";
