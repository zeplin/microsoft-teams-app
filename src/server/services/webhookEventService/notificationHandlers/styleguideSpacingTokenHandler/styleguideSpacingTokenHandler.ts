import {
    StyleguideSpacingTokenCreateEvent,
    StyleguideSpacingTokenUpdateEvent,
    WebhookEvent
} from "../../../../adapters/zeplin/types";
import { NotificationHandler } from "../NotificationHandler";
import { SHORT_DELAY } from "../constants";
import { commonTeamsCard, MessageCard } from "../teamsCardTemplates";
import { ZEPLIN_WEB_APP_BASE_URL, ZEPLIN_MAC_APP_URL_SCHEME } from "../../../../config";
import { URL } from "url";
import { getMacAppRedirectURL } from "../getMacAppRedirectURL";
import { md } from "../md";
import { getRandomEmoji } from "../getRandomEmoji";

type Event = StyleguideSpacingTokenCreateEvent | StyleguideSpacingTokenUpdateEvent;

class StyleguideSpacingTokenHandler extends NotificationHandler<Event> {
    delay = SHORT_DELAY;
    private getText(events: Event[]): string {
        const [{
            payload: {
                action,
                context: {
                    styleguide: {
                        name: styleguideName
                    }
                },
                resource: {
                    data: {
                        name: pivotSpacingTokenName
                    }
                }
            }
        }] = events;
        const actionText = action === "created" ? "added" : "updated";
        return events.length === 1
            ? md`**${pivotSpacingTokenName}** is ${actionText} in _${styleguideName}_! ${getRandomEmoji()}`
            : md`**${events.length} spacing tokens** are ${actionText} in _${styleguideName}_! ${getRandomEmoji()}`;
    }

    private getWebappURL(
        events: Event[]
    ): string {
        const [{
            payload: {
                context: {
                    styleguide: {
                        id: styleguideId
                    }
                }
            }
        }] = events;
        const webappURL = new URL(ZEPLIN_WEB_APP_BASE_URL);
        webappURL.pathname = `styleguide/${styleguideId}/spacing`;
        events.forEach(event => webappURL.searchParams.append("sptid", event.payload.resource.id));
        return webappURL.toString();
    }

    private getMacAppURL(
        events: Event[]
    ): string {
        const [{
            payload: {
                context: {
                    styleguide: {
                        id: styleguideId
                    }
                }
            }
        }] = events;
        return getMacAppRedirectURL(`${ZEPLIN_MAC_APP_URL_SCHEME}://spacing?stid=${styleguideId}&sptids=${events.map(event => event.payload.resource.id).join(",")}`);
    }

    shouldHandleEvent(event: WebhookEvent): event is Event {
        return event.payload.action !== "deleted";
    }

    getTeamsMessage(
        events: Event[]
    ): MessageCard {
        return commonTeamsCard({
            text: this.getText(events),
            section: {
                text: "Make sure your stylesheets are up to date!"
            },
            links: [{
                title: "Open in App",
                url: this.getMacAppURL(events)
            }, {
                title: "Open in Web",
                url: this.getWebappURL(events)
            }]
        });
    }
}

export const styleguideSpacingTokenHandler = new StyleguideSpacingTokenHandler();
