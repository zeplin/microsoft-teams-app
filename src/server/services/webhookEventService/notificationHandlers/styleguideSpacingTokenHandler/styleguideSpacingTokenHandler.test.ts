import { StyleguideSpacingTokenCreatedEvent, StyleguideSpacingTokenUpdatedEvent } from "@zeplin/sdk";

import { StyleguidePlatformEnum } from "../../../../enums";

import { styleguideSpacingTokenHandler } from "./styleguideSpacingTokenHandler";

type GetDummyEventParams = {
    action?: string;
    spacingTokenId?: string;
    spacingTokenName?: string;
    styleguidePlatform?: string;
}

function getDummyEvent({
    action = "created",
    spacingTokenId = "spacingTokenId",
    spacingTokenName = "spacingTokenName",
    styleguidePlatform = StyleguidePlatformEnum.WEB
}: GetDummyEventParams = {}): StyleguideSpacingTokenCreatedEvent | StyleguideSpacingTokenUpdatedEvent {
    return {
        action,
        context: {
            styleguide: {
                id: "styleguideId",
                name: "styleguideName",
                platform: styleguidePlatform
            }
        },
        resource: {
            id: spacingTokenId,
            data: {
                id: spacingTokenId,
                name: spacingTokenName
            }
        }
    } as StyleguideSpacingTokenCreatedEvent | StyleguideSpacingTokenUpdatedEvent;
}

describe("styleguideSpacingTokenHandler", () => {
    describe("getTeamsMessage method", () => {
        beforeAll(() => {
            jest.spyOn(Math, "random").mockReturnValue(0);
        });

        describe("for created notification", () => {
            it("should match snapshot when there is only 1 event", () => {
                expect(
                    styleguideSpacingTokenHandler.getTeamsMessage([getDummyEvent()])
                ).toMatchSnapshot();
            });

            it("should match snapshot when there are more than 1 events", () => {
                expect(
                    styleguideSpacingTokenHandler.getTeamsMessage([
                        getDummyEvent(),
                        getDummyEvent({
                            spacingTokenId: "anotherSpacingTokenId",
                            spacingTokenName: "anotherSpacingTokenName"
                        })
                    ])
                ).toMatchSnapshot();
            });

            it.each([
                StyleguidePlatformEnum.ANDROID,
                StyleguidePlatformEnum.IOS,
                StyleguidePlatformEnum.MAC_OS,
                StyleguidePlatformEnum.WEB,
                StyleguidePlatformEnum.BASE
            ])("should match snapshot when styleguide platform is %s",
                styleguidePlatform => {
                    expect(
                        styleguideSpacingTokenHandler.getTeamsMessage([getDummyEvent({ styleguidePlatform })])
                    ).toMatchSnapshot();
                }
            );
        });

        describe("for updated notification", () => {
            it("should match snapshot when there is only 1 event", () => {
                expect(
                    styleguideSpacingTokenHandler.getTeamsMessage([getDummyEvent({ action: "updated" })])
                ).toMatchSnapshot();
            });

            it("should match snapshot when there are more than 1 events", () => {
                expect(
                    styleguideSpacingTokenHandler.getTeamsMessage([
                        getDummyEvent({
                            action: "updated"
                        }),
                        getDummyEvent({
                            action: "updated",
                            spacingTokenId: "anotherSpacingTokenId",
                            spacingTokenName: "anotherSpacingTokenName"
                        })
                    ])
                ).toMatchSnapshot();
            });

            it.each([
                StyleguidePlatformEnum.ANDROID,
                StyleguidePlatformEnum.IOS,
                StyleguidePlatformEnum.MAC_OS,
                StyleguidePlatformEnum.WEB,
                StyleguidePlatformEnum.BASE
            ])("should match snapshot when styleguide platform is %s",
                styleguidePlatform => {
                    expect(
                        styleguideSpacingTokenHandler.getTeamsMessage([getDummyEvent({ styleguidePlatform })])
                    ).toMatchSnapshot();
                }
            );
        });
    });
});
