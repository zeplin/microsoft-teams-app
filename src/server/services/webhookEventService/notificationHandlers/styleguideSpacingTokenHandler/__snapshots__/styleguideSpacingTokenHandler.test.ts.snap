// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styleguideSpacingTokenHandler getTeamsMessage method for created notification should match snapshot when styleguide platform is android 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure \`spacing.xml\` is up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for created notification should match snapshot when styleguide platform is base 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for created notification should match snapshot when styleguide platform is ios 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for created notification should match snapshot when styleguide platform is macos 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for created notification should match snapshot when styleguide platform is web 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for created notification should match snapshot when there are more than 1 events 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**2 spacing tokens** are added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%2CanotherSpacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&sptid=anotherSpacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for created notification should match snapshot when there is only 1 event 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for updated notification should match snapshot when styleguide platform is android 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure \`spacing.xml\` is up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for updated notification should match snapshot when styleguide platform is base 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for updated notification should match snapshot when styleguide platform is ios 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for updated notification should match snapshot when styleguide platform is macos 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for updated notification should match snapshot when styleguide platform is web 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is added to _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for updated notification should match snapshot when there are more than 1 events 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**2 spacing tokens** are updated in _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%2CanotherSpacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&sptid=anotherSpacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;

exports[`styleguideSpacingTokenHandler getTeamsMessage method for updated notification should match snapshot when there is only 1 event 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "**spacingTokenName** is updated in _styleguideName_! ðŸ—ž",
  "sections": [
    {
      "text": "Make sure your tokens are up to date!",
      "images": []
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://msteams-app.zeplin.io/api/app-redirect?uri=zpl%3A%2F%2Fspacing%3Fstid%3DstyleguideId%26sptids%3DspacingTokenId%26ref%3Dmicrosoft-teams"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://app.zeplin.io/styleguide/styleguideId/spacing?sptid=spacingTokenId&ref=microsoft-teams"
        }
      ]
    }
  ]
}
`;
